### DISCUSSION TITLE
external converter with unsoppetet Tuya "_TZE200_kly8gjlz"

### DISCUSSION THREAD

**Discussion by fardem** [Koenkk/zigbee2mqtt#9868] [Devices]
Hi i bought a thermostat from aliexpress:
https://de.aliexpress.com/item/1005003448958272.html?spm=a2g0s.9042311.0.0.2ecc4c4dFgHd0E

This is not supported by the current zigbe2mqtt. So I was looking for an external converter.
And found this one.
https://gist.github.com/geoffreylagaisse/bb90214dbaf687465e5841f49685cf66

So far this converter works best with the Theromstat compared to others.
However, the thermostat behaves strangely. When I switch the preset in the Zigbee2mqtt and also in the Homassitant. For example, from Shedule to Manual this jumps once to "Auto" and only then to "Heat" and the set temperature is overwritten with 20 ° C and remains at 20 ° C. 
But the same thing also happens if, when in heat mode, I simply press heat again 

As a result, I cannot automate the temperature control with a scheduler.

I can't find the cause of the jump in the code. Where can I find the cause.

With other i have not this problem but the  temperature calibration  dont work and also not the display for the local temperature does not work.

So I would like to get the one that works best so far to work properly.

**Comment by ypoosn** [comment]
Hi,

did you check this?

[https://github.com/Koenkk/zigbee-herdsman-converters/issues/3142](url)

Display elements look the same rotated by 180 degrees.
Maybe its behavior is identical to TS0601 _TZE200_hue3yfsn ?

best, Ypo

**Reply by fardem** [reply]
> did you check this?
> 
> [https://github.com/Koenkk/zigbee-herdsman-converters/issues/3142](url)

the link is not reachable

**Reply by ypoosn** [reply]
> the link is not reachable

Sorry pasting the link with add link button somehow did not work. I tried again below. If this does not work you have to 
copy the url as text and paste it into your address bar.

https://github.com/Koenkk/zigbee-herdsman-converters/issues/3142

best, ypo

**Reply by fardem** [reply]
Hello, thank you,

With the external converter, I can now use the scheduler in the homeassistant. The jumping is now gone and it also takes the temperature input.

There are still some things that don't work. For example you cannot turn off the thermostat, or adjust the scheduler
But I will try to understand the code from the external converter and adapt it to the thermostat.
